cmd/moby/config.go

```diff
@@ -185,39 +257,193 @@ func (m mlist) parts(i int) int {
 	return strings.Count(filepath.Clean(m[i].Destination), string(os.PathSeparator))
 }
 
-// ConfigInspectToOCI converts a config and the output of image inspect to an OCI config file
-func ConfigInspectToOCI(image *MobyImage, inspect types.ImageInspect) ([]byte, error) {
-	oci := specs.Spec{}
+// assignBool does ordered overrides from JSON bool pointers
+func assignBool(v1, v2 *bool) (ret bool) {
+	if v1 != nil {
+		ret = *v1
+	}
+	if v2 != nil {
+		ret = *v2
+	}
+	return
+}
 
-	config := inspect.Config
-	if config == nil {
-		return []byte{}, errors.New("empty image config")
+// assignBoolPtr does ordered overrides from JSON bool pointers
+func assignBoolPtr(v1, v2 *bool) (ret *bool) {
+	if v1 != nil {
+		ret = v1
+	}
+	if v2 != nil {
+		ret = v2
 	}
+	return
+}
 
-	args := append(config.Entrypoint, config.Cmd...)
-	if len(image.Command) != 0 {
-		args = image.Command
+// assignIntPtr does ordered overrides from JSON int pointers
+func assignIntPtr(v1, v2 *int) (ret *int) {
```

Comment:
The intended behavior If v2 and v1 are both set is to return v2 right?  Wouldn't be better to early return doing something like:

```go
	if v2 != nil {
		return v2
	}

 	return v1
```

Not everyone feels comfortable with named returns